### PR TITLE
Add optional badge to Catalog Category label

### LINKED ISSUE #5146: Extension API for Catalog Category badge
Allow extension to specify a badge component to be displayed after the Category label in the Catalog Category menu.

Example usage in extension:
```
export default class extends Renderer.LensExtension {
  customCategoryViews = [
    {
      group: "entity.k8slens.dev",
      kind: "KubernetesCluster",
      priority: 10,
      components: {
        View: MyKubernetesClusterView,
        Badge: MyBadge
      },
    },
  ];
}
```

### PR DESCRIPTION
**Pull request by panuhorsmalahti**
* Fixes https://github.com/lensapp/lens/issues/5146
* Makes it possible for extensions to specify a badge for a custom Catalog Category
* Add `badgeBackgroundColor` CSS variable for extension use

Example use by an extension:
Screenshot 2022-03-31 at 16 17 35

### CODE REVIEW

File: src/renderer/components/+catalog/catalog-category-label.tsx
Comment (jweak) [inline src/renderer/components/+catalog/catalog-category-label.tsx]: I think it's better not to render container div if there's no badge.

We could also have props like below, so this UI component wouldn't need to know about category interface.
```react.jsx
{
  label: string | React.ReactNode,
  badge?: React.ReactNode
}
```

Comment (panuhorsmalahti) [reply src/renderer/components/+catalog/catalog-category-label.tsx]: Good points, changed

**Review by jweak — DISMISSED** [review]

File: src/renderer/components/+catalog/catalog-category-label.tsx
Comment (Nokel81) [inline src/renderer/components/+catalog/catalog-category-label.tsx]: I think that this should take only a `Category` as its input.

Comment (panuhorsmalahti) [reply src/renderer/components/+catalog/catalog-category-label.tsx]: This was discussed here before it was changed to take in just label and badge: https://github.com/lensapp/lens/pull/5156#discussion_r839652665

Comment (Nokel81) [reply src/renderer/components/+catalog/catalog-category-label.tsx]: Interesting, I agree with the first part of that comment, but not really the second since this component is specifically about doing all that logic, I don't see the benefit of making the caller provide it.

Comment (Nokel81) [reply src/renderer/components/+catalog/catalog-category-label.tsx]: @jweak Thoughts?

Comment (jweak) [reply src/renderer/components/+catalog/catalog-category-label.tsx]: I think a presentational component like Label should not do that logic.

Comment (Nokel81) [reply src/renderer/components/+catalog/catalog-category-label.tsx]: Interesting, may I garner some insight then. 

 would have said that a presentation component was the best place to do it because it encapsulates the behaviour. The user of this component doesn't care if it displays or not.

Comment (jweak) [reply src/renderer/components/+catalog/catalog-category-label.tsx]: I can see both implementations have merit. My point of would be to have a numbered components like `<Label>foo</Label>` and `<Header size={2}>catalog</Header>` and not create a new Label or Header component for each object we would like to display something.  

Anyway, if you feel strongly that it's better the other way, I'm fine with changing it back. It was just a suggestion.

Comment (Nokel81) [reply src/renderer/components/+catalog/catalog-category-label.tsx]: I don't feel terribly strong about it, I just wanted to know you thought pattern.

I agree 100% that there should be some base component for doing the some rendering but I also think that there should be a separate component for doing the logic (and not having it be done in line).

Comment (jim-docker) [reply src/renderer/components/+catalog/catalog-category-label.tsx]: I think the name `CatalogCategoryLabel` implies it is specific to a catalog category and it seems appropriate to pass just a `Category` to it. If it were called something generic like `LabelWithBadge` and could be used more generally then passing the `label` and `badge` are appropriate

**Review by jim-docker — DISMISSED** [review]
tested from extension

File: src/renderer/components/+catalog/catalog-category-label.tsx
Comment (panuhorsmalahti) [reply src/renderer/components/+catalog/catalog-category-label.tsx]: Waiting for 1 more approval.

Comment (jweak) [reply src/renderer/components/+catalog/catalog-category-label.tsx]: I guess it's two against one for changing it back to taking the category so maybe change it and let's approve this and get it merged.

Comment (panuhorsmalahti) [reply src/renderer/components/+catalog/catalog-category-label.tsx]: It's changed back now

**Review by jweak — APPROVED** [review]

**Review by Nokel81 — APPROVED** [review]

**Pull request merged**
